build-logic/gradle/src/main/kotlin/com/avito/android/GenerateCommonProperties.kt

```diff
@@ -0,0 +1,55 @@
+package com.avito.android
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectories
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+abstract class GenerateCommonProperties : DefaultTask() {
```

Comment:
- [ ] Add gradleTests please
- [ ] Add execution on each CI build
- [ ] Should we make task cacheble?

Reply:
Gradle tests should be done for sure, but can't use our TestProjectGenerator directly here
Options are:
 - add it as mavenCentral dependency
 - write test on pure GradleTestKit

[For future self]: don't forget that these tests won't be started on CI, need to wire it additionally

Reply:
Execution on every build is tricky.
I have to add additional gradle build for that, because gradle.properties changes will affect build only on second run.
This is the same situation as gradle wrapper task calling. 
So probably it is not a good idea

Reply:
Yep, we can make this task cacheable, but it won't be much profitable if task will be executed only manually (see previous comment)

Reply:
> but can't use our TestProjectGenerator directly
This is a problem. I don't know what is better but prefer `add it as mavenCentral dependency`
> Execution on every build is tricky.
But what If I add some properties and forget to regenerate them?
How will we check the correctness and integration of `gradle.properties` that should be generated?
I don't want to think about that I shouldn't forget to regenerate something. I want CI help me with this

Reply:
The only viable solution i see is to check if all properties have correct values from `common` properties file, and ask to regenerate it.
Because it should be under version control and we don't have ability to commit and push via CI yet (and maybe for good)